src/TestApps/TestApp.NET45/Client.cs

```diff
@@ -44,21 +44,32 @@ public void Dispose()
                         //if (r.Uri == "/BinarySwitchResURI" || r.Uri == "/light/1")
                         {
                             var client = new IotivityDotNet.ResourceClient(e.Address, r.Uri);
-                            //Get all the properties from the resource
-                            var response = await client.GetAsync(type);
-                            var p = response.Payload as IotivityDotNet.RepPayload;
-                            while(p != null)
+                            try
                             {
-                                foreach(var item in p.Values)
+                                //Get all the properties from the resource
+                                var response = await client.GetAsync(type);
+
+
+                                var p = response.Payload as IotivityDotNet.RepPayload;
+                                while (p != null)
                                 {
-                                    Console.WriteLine($"\t\t\t\t{item.Key} = {item.Value}");
+                                    foreach (var item in p.Values)
+                                    {
+                                        Console.WriteLine($"\t\t\t\t{item.Key} = {item.Value}");
+                                    }
+                                    p = p.Next;
                                 }
-                                p = p.Next;
-                            }
 
-                            //Start observing the resource
-                            client.OnObserve += OnResourceObserved;
-                        }
+                                //Start observing the resource
+                                client.OnObserve += OnResourceObserved;
```

Comment:
I agree that the error should still be fixed (I looked a bit into it in #3) but I think this code can be moved out of the try-catch block
```
//Start observing the resource
client.OnObserve += OnResourceObserved;
```